compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -1335,6 +1357,16 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
     /** Suppress further inlining if this inline typer has already issued errors */
     override def suppressInline(using Context) =
       ctx.reporter.errorCount > initialErrorCount || super.suppressInline
+
+    private def inlineIfIsNestedInlineCall(tree: Tree)(using Context): Tree =
```

Comment:
This looks like a change in behavior previously we just returned the call and Typer would do the re-inlining. Why is that now done here? Typer already contains the logic to do repeated inline calls in apply. Should there be a similar call in Inliner? I.e. a single toplevel call, instead of 4 in different specific places?

Reply:
The issue is that when we are inlining in the `Inlining` phase there is no normal typer to inline those nested calls.

Reply:
I will try to see if it is possible to have a single place to inline nested calls.

Reply:
Refactored the code in 53d69cdeb54111ce5b23c6a70ccb9b87016496b9

Reply:
@odersky could you review that last commit before merging?